bokeh/application/handlers/notebook.py

```diff
@@ -32,17 +32,53 @@ def __init__(self, *args, **kwargs):
             filename (str) : a path to a Jupyter notebook (".ipynb") file
 
         '''
-        nbformat = import_required('nbformat', 'The Bokeh notebook application handler requires Jupyter Notebook to be installed.')
-        nbconvert = import_required('nbconvert', 'The Bokeh notebook application handler requires Jupyter Notebook to be installed.')
-
         if 'filename' not in kwargs:
             raise ValueError('Must pass a filename to NotebookHandler')
         filename = kwargs['filename']
 
+        preprocessors = self._preprocessors()
+        try:
+            source, meta = self.export_python(filename, preprocessors)
+        except:
+            # If any preprocessors resulted in an exception, convert without them
+            source, meta = self.export_python(filename, preprocessors=[])
```

Comment:
I the try/except necessary? This seems like it might silently cover up errors, or obscure the real source of problems.